xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java

```diff
@@ -229,32 +210,39 @@ public void startAndStopCanBeCalledMultipleTimes() {
     verifyNoMoreInteractions(requestObserver);
 
     lrsClient.stopLoadReporting();
-    verify(requestObserver).onCompleted();
+    assertThat(callEnded.get()).isTrue();
     assertThat(fakeClock.getPendingTasks(LRS_RPC_RETRY_TASK_FILTER)).isEmpty();
     lrsClient.stopLoadReporting();
-    verifyNoMoreInteractions(requestObserver);
+    assertThat(callEnded.get()).isTrue();
 
     lrsClient.startLoadReporting(callback);
     verify(mockLoadReportingService, times(2)).streamLoadStats(lrsResponseObserverCaptor.capture());
     assertThat(lrsRequestObservers).hasSize(2);
   }
 
+  // Currently we expect each gRPC client talks to a single service per cluster, so we test LRS
+  // client reporting load for a single cluster service only.
+  // TODO(chengyuanzhang): Existing test suites for LRS client implementation have poor behavior
+  //  coverage and are not robust. Should improve once its usage is finalized without too much
+  //  assumption.
+
   @Test
   public void loadReportActualIntervalAsSpecified() {
     verify(mockLoadReportingService).streamLoadStats(lrsResponseObserverCaptor.capture());
     StreamObserver<LoadStatsResponse> responseObserver = lrsResponseObserverCaptor.getValue();
     assertThat(lrsRequestObservers).hasSize(1);
     StreamObserver<LoadStatsRequest> requestObserver = lrsRequestObservers.poll();
-    when(loadStatsStore.generateLoadReport()).thenReturn(ClusterStats.newBuilder().build());
-    InOrder inOrder = inOrder(requestObserver, loadStatsStore);
+
+    // Add load stats source for some cluster service.
+    when(mockLoadStatsStore.generateLoadReport()).thenReturn(ClusterStats.newBuilder().build());
+    lrsClient.addLoadStatsStore("namespace-foo:service-blade", mockLoadStatsStore);
+
+    InOrder inOrder = inOrder(requestObserver, mockLoadStatsStore);
```

Comment:
The comment regarding putting `mockLoadStatsStore` into `InOrder` goes away due to a force-push (apologize for that).

I do orderly verify invocation of the `LoadStatsStore` instance with `requestObserver` in `assertNextReport(...)` method. I refactored that helper method to take in the `LoadStatsStore` instance to make this clearer.

`LoadReportClientImplTest.java` contains some badly written tests (I put a note at line 225) and some bad programming practices (or even something that was incorrect) that I did not realize until recently. I am fixing those that are significantly bad/incorrect. But possibly, still things look quite inappropriate.